### PR TITLE
[tools][Android] Add android-update-native-dependencies command

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

Based on #11919 with review fixes, more context is in that PR (I can't reopen #11919). 

Also changed version selection to make updating to latest the default option and avoid retyping numbers:
image
image

# How

Moved to `gradle` `initScript` flag from static dependencies.

### CODE REVIEW

File: tools/src/android-update-native-dependencies/prompts.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  projectName,
+  gradleReport: { outdated, exceeded, unresolved },
+}: AndroidProjectReport) {
+  const name = `${projectName}${
+    outdated.length > 0 ? ` ${chalk.yellow(`(${outdated.length} ⚠️ )`)}` : ''
+  }${
+    exceeded.length > 0 || unresolved.length > 0
+      ? ` ${chalk.red(`(${exceeded.length + unresolved.length} ❗️)`)}`
+      : ''
+  }`;
```
Comment (tsapeta) [inline tools/src/android-update-native-dependencies/prompts.ts:24]: This doesn't look well, could you maybe move these interpolated values to separate variables?

File: tools/src/android-update-native-dependencies/prompts.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  ]);
+  return reports.filter(({ projectName }) => selectedProjects.includes(projectName));
+}
+
+async function promptForDependenciesVersions(
+  dependencies: GradleDependency[]
+): Promise<GradleDependencyUpdate[]> {
+  const updates: GradleDependencyUpdate[] = [];
+
+  for (const dependency of dependencies.sort((a, b) => a.fullName.localeCompare(b.fullName))) {
```
Comment (tsapeta) [inline tools/src/android-update-native-dependencies/prompts.ts:56]: Let's sort this array outside the loop 😅

File: tools/src/android-update-native-dependencies/prompts.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          {
+            type: 'input',
+            name: 'version',
+            message: `${dependency.fullName}:${dependency.currentVersion} ➡️ `,
+            default: addColorBasedOnSemverDiff(dependency.availableVersion, semverDiff),
+            prefix: `  ${chalk.green('?')}`,
+          },
+        ])
+      ).version;
+    }
```
Comment (tsapeta) [inline tools/src/android-update-native-dependencies/prompts.ts:104]: Hmm, this deserves for a separate function 😉

File: tools/src/android-update-native-dependencies/updateChangelogFiles.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          Changelogs.UNPUBLISHED_VERSION_NAME,
+          Changelogs.ChangeType.LIBRARY_UPGRADES,
+          null,
+          updates.map(
+            (update) =>
+              `Updated Android native dependency: ${maybeWrapInChangelogLink(
+                `\`${update.fullName}:${update.oldVersion}\` ➡️ \`${update.newVersion}\``,
+                update.fullName
+              )}`
+          )
```
Comment (tsapeta) [inline tools/src/android-update-native-dependencies/updateChangelogFiles.ts:30]: I think this should be done outside the `insertEntriesAsync` invocation — if a single argument takes more than one line, it starts disrupting the reader to understand the flow.

File: tools/src/android-update-native-dependencies/updateGradleFiles.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    `${name}('|"),(\\s+)?version:(\\s+)('|")${oldVersion}('|")`
+  );
+  regexDependencyWithVersionPrefix
+    .exec(modifiedBody)
+    ?.filter((it) => it.includes(oldVersion))
+    .forEach((match) => {
+      modifiedBody = modifiedBody.replace(match, match.replace(oldVersion, newVersion));
+    });
+
+  return modifiedBody;
```
Comment (tsapeta) [inline tools/src/android-update-native-dependencies/updateGradleFiles.ts:92]: I'll need to read this function more carefully when I'm back 😅 I feel like it's pretty overcomplicated and probably we could use some helpers for file transformations that we already have (e.g. versioning scripts). Maybe @lukmccall could help you when you both are at the office.

Comment (aleqsio) [reply tools/src/android-update-native-dependencies/updateGradleFiles.ts:92]: We decided to keep as is after all

**Review by tsapeta — APPROVED** [review]

**Review by lukmccall — APPROVED** [review]

**Pull request merged**